agent/consul/server_test.go

```diff
@@ -1129,6 +1133,126 @@ func TestServer_RPC(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	storage := make(map[string]float32)
+	keyMakingFunc := func(key []string) string {
+		return strings.Join(key, "+")
+	}
+
+	simpleRecorderFunc := func(key []string, val float32, labels []metrics.Label) {
+		storage[keyMakingFunc(key)] = val
+	}
+
+	_, conf := testServerConfig(t)
+
+	// "disable" metrics net/rpc interceptor
+	conf.GetNetRPCInterceptorFunc = nil
+
+	s, err := newServer(t, conf)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	defer s.Shutdown()
+	testrpc.WaitForTestAgent(t, s.RPC, "dc1")
+
+	// "hijack" the rpc recorder for asserts;
+	// note that there will be "internal" net/rpc calls made
+	// that will still show up; those don't go thru the net/rpc interceptor;
+	// see consul.agent.rpc.middleware.RPCTypeInternal for context
+	s.rpcRecorder = &middleware.RequestRecorder{
+		Logger:       hclog.NewInterceptLogger(&hclog.LoggerOptions{}),
+		RecorderFunc: simpleRecorderFunc,
+	}
+
+	t.Run("test no net/rpc interceptor metric", func(t *testing.T) {
+		var out struct{}
+		if err := s.RPC("Status.Ping", struct{}{}, &out); err != nil {
+			t.Fatalf("err: %v", err)
+		}
+
+		key := keyMakingFunc(middleware.OneTwelveRPCSummary[0].Name)
+
+		if _, ok := storage[key]; ok {
+			t.Fatalf("Did not expect to find key %s in the metrics log, ", key)
+		}
+	})
+}
+
+func TestServer_RPC_MetricsIntercept(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	_, conf := testServerConfig(t)
+
+	// The method used to record metric observations here is similar to that used in
+	// interceptors_test.go; at present, we don't have a need to lock yet but if we do
+	// we can imitate that set up further or just factor it out as a "mock" metrics backend
+	storage := make(map[string]float32)
+	keyMakingFunc := func(key []string, labels []metrics.Label) string {
+		allKey := strings.Join(key, "+")
+
+		for _, label := range labels {
+			allKey = allKey + "+" + label.Value
+		}
+
+		return allKey
+	}
+
+	simpleRecorderFunc := func(key []string, val float32, labels []metrics.Label) {
+		storage[keyMakingFunc(key, labels)] = val
+	}
+	actualRecorder := &middleware.RequestRecorder{
+		Logger:       hclog.NewInterceptLogger(&hclog.LoggerOptions{}),
+		RecorderFunc: simpleRecorderFunc,
+	}
+
+	conf.GetNetRPCInterceptorFunc = func(recorder *middleware.RequestRecorder) rpc.ServerServiceCallInterceptor {
+		// we re-assign our recorder in here
+		recorder = actualRecorder
+
+		return func(reqServiceMethod string, argv, replyv reflect.Value, handler func() error) {
+			reqStart := time.Now()
+
+			err := handler()
+
+			recorder.Record(reqServiceMethod, "test", reqStart, argv.Interface(), err != nil)
+		}
+	}
+
+	s, err := newServer(t, conf)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	defer s.Shutdown()
+	testrpc.WaitForTestAgent(t, s.RPC, "dc1")
+
+	// asserts
+	t.Run("test happy path for metrics interceptor", func(t *testing.T) {
```

Comment:
* see `interceptors_test.gp` for "data" level and unit level testing
* see `metrics_test.go` for e2e/ bdd testing

all in main already